### ISSUE TITLE
Running inference without gradio

### ISSUE THREAD

**Issue by fradif96** [cvlab-columbia/zero123#61]
Hi, 

Thank you for the fantastic work.
I would like to ask if there is a way to run inference without gradio.
My goal is to give a desired pose for the diffusion guidance and just obtain a novel view of the image saved in some dir. 

Thank you in advance

**Comment by jwlee-vcl** [comment]
I haven't tested this code.
good luck :)

[predict.zip](https://github.com/cvlab-columbia/zero123/files/12068765/predict.zip)

**Comment by fradif96** [comment]
Thank you very much!
I will test it asap

**Comment by HaoyangHuang** [comment]
> Thank you very much! I will test it asap

can you inference with this file? thanks!

**Comment by fradif96** [comment]
Hello 
Yes the code works fine. 
Just on the line in which the concatenation is performed 
`c = torch.cat([c, T], dim=-1)
`
should be substituted by 
`c = torch.cat([c, T], dim=-1).float()
`
In my case it throws an error for type mismatching.

[predict.zip](https://github.com/cvlab-columbia/zero123/files/12121861/predict.zip)

**Comment by yesterday-Luo** [comment]
Could you tell me if the code can generate new views for a batch images?  My goal is to give a batch images and a batch poses, and generate one view for every image from a batch.

**Comment by fradif96** [comment]
@yesterday-Luo 
The code is meant to generate one novel view given one image and one pose. 
However, I think that it should be straightforward to adapt it for your use case, by just creating a for-loop for the batch of images-poses.
